File: backend/Models/Label.cs
using System.ComponentModel.DataAnnotations;

namespace DataLabellingSupportSystem.Api.Models;

public class Label
{
    [Required]
    [MaxLength(24)]
    public string Id { get; set; } = string.Empty;

    [Required]
    [MaxLength(24)]
    public string ProjectId { get; set; } = string.Empty;

    public Project? Project { get; set; }

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [MaxLength(24)]
    public string? CategoryId { get; set; }

    public LabelCategory? Category { get; set; }

    [MaxLength(24)]
    public string? AnnotationTypeId { get; set; }

    public AnnotationTypeDefinition? AnnotationType { get; set; }

    public int YoloClassId { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }
}

File: mobile_app/lib/models/annotator/label.dart
class Label {
  final String id;
  final String name;
  final String color;
  final String projectId;
  final int yoloClassId;

  Label({
    required this.id,
    required this.name,
    required this.color,
    required this.projectId,
    required this.yoloClassId,
  });

  factory Label.fromJson(Map<String, dynamic> json) {
    return Label(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      color: json['color'] ?? '#2563eb',
      projectId: json['projectId'] ?? '',
      yoloClassId: json['yoloClassId'] ?? 0,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'color': color,
        'projectId': projectId,
        'yoloClassId': yoloClassId,
      };
}
